utxoscanner.go

```diff
@@ -0,0 +1,515 @@
+package neutrino
+
+import (
+	"container/heap"
+	"sync"
+	"sync/atomic"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/roasbeef/btcd/chaincfg/chainhash"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+	"github.com/roasbeef/btcutil/gcs/builder"
+	"github.com/roasbeef/btcwallet/waddrmgr"
+)
+
+// GetUtxoRequest is a request to scan for OutPoint from the height StartHeight.
+type GetUtxoRequest struct {
+	// OutPoint is the target outpoint to watch for spentness.
+	OutPoint *wire.OutPoint
+
+	// StartHeight is the height we should begin scanning for spends.
+	StartHeight uint32
+
+	// Result is callback returning either a spend report or an error.
+	//
+	// NOTE: The spend report WILL be nil if the target outpoint could not
+	// be found or was not spent above the start height.
+	Result func(*SpendReport, error)
+}
+
+// UtxoScannerConfig exposes configurable methods for interacting with the blockchain.
+type UtxoScannerConfig struct {
```

Comment:
Currently yes they are all fns, though I think it's likely that this will be amended with other non-fn variables. I've found the ability to piecemeal the functions super convenient for testing and mocking out functions